velox/exec/MergeJoin.cpp

```diff
@@ -96,22 +96,41 @@ void MergeJoin::initializeFilter(
   auto numFields = filter_->expr(0)->distinctFields().size();
   names.reserve(numFields);
   types.reserve(numFields);
+
+  // When 'channel' in the 'filter_' is found in 'inputType', first try to add
```

Comment:
> When 'channel' in the 'filter_' is found in 'inputType',

This sounds strange since it suggests that filter columns may or may not be present in the input. Consider,

```
This function is called in a loop for all columns used in a filter expression. For each column, 'channel' specifies column index in the left or right side of the join, 'outputs' specifies the mapping from input columns of that side of the join to the output of the join; 'filters' specifies the mapping from input columns to the 'filterInput_' for columns that are not projected to the output.

This function checks whether the column is projected to the output and if so adds an entry to 'filterInputToOutputChannel_'. Otherwise, adds an entry to 'filters'.

At the end of the loop each column used in a filter appears either in filterInputToOutputChannel_ (in case the column is projected to the join output), filterLeftInputs_ (in case the column is not projected to the join output and it comes from the left side of the join), filterRightInputs_ ((in case the column is not projected to the join output and it comes from the right side of the join).
```

Reply:
Nice and detailed comment for the function, have updated, thanks!